index.html

```diff
@@ -4292,6 +4306,489 @@ <h3>Proof Formats</h3>
       </section>
     </section>
 
+    <section class="normative">
+      <h2>Algorithms</h2>
+
+      <p>
+This section contains algorithms that can be used by implementations to
+perform common operations.
+      </p>
+
+      <p class="note" title="Implementers can include additional checks, warnings, and errors.">
+Implementers are advised that the algorithms in this section contain the bare
+minimum set of checks used by implementations to test conformance to this
+specification. Implementations are expected to provide additional checks that
+report helpful warnings for developers to help debug potential issues.
+Similarly, implementations are likely to provide additional checks that
+could result in new types of errors being reported in order to stop harmful
+content. Any of these additional checks might be integrated into future
+versions of this specification.
+      </p>
+
+      <section class="normative">
+        <h3>Verification</h3>
+
+        <p>
+This section contains an algorithm that <a>conforming verifier
+implementations</a> MUST run when verifying a <a>verifiable credential</a> or a
+<a>verifiable presentation</a>. This algorithm takes a byte sequence (bytes
+<var>inputBytes</var>) and a media type (string <var>inputMediaType</var>) as
+inputs, and produces an object (<var>result</var>) that contains the
+following:
+        </p>
+
+        <ul>
+          <li>
+a status (boolean <var>status</var>)
+          </li>
+          <li>
+a conforming document (object <var>document</var>)
+          </li>
+          <li>
+a media type (string <var>mediaType</var>), including any media type parameters
+          </li>
+          <li>
+zero or more warnings (array of <a>ProblemDetails</a> <var>warnings</var>)
+          </li>
+          <li>
+zero or more errors (array of <a>ProblemDetails</a> <var>errors</var>)
+          </li>
+        </ul>
+
+        <p>
+The verification algorithm is as follows:
+        </p>
+
+        <ol class="algorithm">
+          <li>
+Set <var>result</var> to an empty object.
+          </li>
+          <li>
+Ensure that the securing mechanism has properly protected the
+<a>conforming document</a> by performing the following steps:
+            <ol class="algorithm">
+              <li>
+Set <var>securingMechanismVerified</var> to the result of executing the
```

Comment:
In my opinion, the VCDM does not define clear or robust enough requirements for securing mechanisms. It would be immensely helpful if we could establish precise requirements for this, as verifiers would then have at least some baseline security guarantees.

The only requirements provided so far state that:
> A securing mechanism MUST express the details necessary to evaluate whether a [verifiable credential](https://www.w3.org/TR/vc-data-model-2.0/#dfn-verifiable-credential) or [verifiable presentation](https://www.w3.org/TR/vc-data-model-2.0/#dfn-verifiable-presentation) has been secured using it.

However, this is vague and not clear enough.

Note that while there is a section on the embedded `proof` property, which is used for data integrity, this does not apply to securing mechanisms that are not part of data integrity. Furthermore, it does not provide sufficiently clear requirements other than stating that it MUST be identified by its `type`.

For reference:
> One or more cryptographic proofs that can be used to detect tampering and verify the authorship of a [verifiable credential](https://www.w3.org/TR/vc-data-model-2.0/#dfn-verifiable-credential) or a [verifiable presentation](https://www.w3.org/TR/vc-data-model-2.0/#dfn-verifiable-presentation). Each proof is a separate [named graph](https://www.w3.org/TR/vc-data-model-2.0/#dfn-named-graphs) (referred to as a proof graph) containing a single proof. The specific method used for an [embedded proof](https://www.w3.org/TR/vc-data-model-2.0/#dfn-embedded-proof) MUST be identified using the type [property](https://www.w3.org/TR/vc-data-model-2.0/#dfn-property). 

This relates to the issue I raised recently: https://github.com/w3c/vc-data-model/issues/1327

Reply:
Note, we can also create a new issue to unblock this PR since I understand I'm already a bit late with this review.

Reply:
At a minimum, it should address data integrity and define the scope of 'data' in data integrity, encompassing external resources, semantics, schemas, etc and how it might relate to other properties in the data model such as `relatedResources`. Additionally, it should cover authorship and what it means if authorship was successfully verified.

Reply:
Alternatively, if securing mechanisms don't have any commonalities by design this should be also stated explicitly, so that verifiers understand that security guarantees are intentionally not equal.

Reply:
I think this is a good idea, but that it should wait for a dedicated PR.

Reply:
+1 this makes sense to me and aligns with what I saw during the PING review of the JOSE/COSE specs. I'm not necessarily tied to making sure this PR is blocked in order to properly specify it though.

Reply:
Raised issue #1374 to track this concern.